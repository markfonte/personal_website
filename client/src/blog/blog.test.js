import React from 'react';
import Blog from './blog.jsx';
import renderer from 'react-test-renderer';

test('Extra matches snapshot', () => {
  const component = renderer.create(
      <Blog/>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
