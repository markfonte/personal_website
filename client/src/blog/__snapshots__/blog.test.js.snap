// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extra matches snapshot 1`] = `
<div>
  <p
    className="MuiTypography-root Blog-title-1 MuiTypography-h6 MuiTypography-paragraph"
  >
    Welcome! Here are some of my thoughts on issues that I find important.
    <br />
    Disagreements and discussions are highly encouraged üòä
  </p>
  <div
    className="MuiPaper-root MuiCard-root card MuiPaper-elevation8 MuiPaper-rounded"
  >
    <div
      className="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-controls="Owning Your Job Negotiations-content"
        aria-disabled={false}
        aria-expanded={false}
        aria-label="Owning Your Job Negotiations"
        className="MuiButtonBase-root MuiAccordionSummary-root"
        id="Owning Your Job Negotiations-header"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content"
        >
          <a
            className="gone"
            href="#Owning Your Job Negotiations_card"
            name="Owning Your Job Negotiations_card"
          >
            Owning Your Job Negotiations
             Card
          </a>
          <div
            className="Blog-expansionHeaderContainer-2"
          >
            <div
              className="MuiCardHeader-root cardHeader"
            >
              <div
                className="MuiCardHeader-content"
              >
                <span
                  className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  Owning Your Job Negotiations
                </span>
              </div>
            </div>
            <h6
              className="MuiTypography-root MuiTypography-subtitle1"
            >
              <br />
              <div>
                Employers have been looking out for their best interests since their conception. Unfortunately, employees often 
                <i>
                  allow themselves to be taken advantage of
                </i>
                 by their employers, particularly during employment negotiations. It‚Äôs time to change that.
              </div>
            </h6>
          </div>
        </div>
        <div
          aria-disabled={false}
          aria-hidden={true}
          className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={null}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <button
              aria-describedby={null}
              className="MuiButtonBase-root MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "padding": 0,
                }
              }
              tabIndex={0}
              title="Expand/collapse card"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        className="MuiCollapse-container MuiCollapse-hidden"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="Owning Your Job Negotiations-header"
              id="Owning Your Job Negotiations-content"
              role="region"
            >
              <div
                className="MuiAccordionDetails-root"
              >
                <div
                  className="MuiCardContent-root Blog-cardContent-3"
                >
                  <p
                    className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-paragraph"
                  >
                    *Disclaimer: I am writing this as a gainfully employed member of Amazon. I am thankful for that job, and I understand the risks of publicly sharing any criticisms of the company for which I work. However, I will not refrain from telling the truth as I see it, so Amazon and other companies mentioned here are free to take my words as the constructive criticism in which they were meant.*
                  </p>
                  <p
                    className="MuiTypography-root MuiTypography-body2 MuiTypography-paragraph"
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1"
                    >
                      <h5
                        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
                      >
                        Companies Looking Out For Their Own Best Interests, Part One
                      </h5>
                      Employment contracts in the United States in 2020 are almost exclusively ‚Äúemployee-at-will.‚Äù Meaning, if the employee wants to quit at any point, they are free to do so. If the employer wants to let the employee go without cause, they are free to do so (unless a legitimate claim can be made that it was an unjust termination of the basis of gender, race, etc.).
                      <br />
                      <br />
                      Why do I bring this up? I often find that people talk about their employment contracts as if they are legally binding documents that force them to work for the company, even if they don‚Äôt want to. In almost every case, this is not true.
                      <br />
                      <br />
                      There are many things that an employer can do to ‚Äúforce the hand‚Äù of potential employees (we will get to these later), and one of them is getting them to believe that by signing a contract to work with a company, they must actually go work for that company. Once you recognize that this is just a subtle tactic to tilt the scales in their favor, you can begin to resist it.
                      <br />
                      <br />
                      <h5
                        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
                      >
                        ‚ÄúExploding Offers‚Äù & The Stigma Around Reneging
                      </h5>
                      Different companies can have very different recruitment schedules - you might apply for a company that offers you an interview a full two months later. For anyone who receives multiple job offers, it stands to reason that the first offer they receive doesn‚Äôt have to be the best offer they can/will receive.
                      <br />
                      <br />
                      In the fall of 2019, I was looking for a full-time software job for when I graduated from college the following May. I started by getting a return offer from J.P. Morgan Chase & Co. at the end of the summer (August 9th, 2019). I was given a deadline of October 1st to accept the offer, or it would be rescinded.
                      <br />
                      <br />
                      Offers like these are called ‚Äúexploding offers‚Äù and are a tool of the oppressor. The problem with an October 1st deadline was that I was currently looking for other job opportunities as well. As someone looking for a job, it is in my best interest to collect as many offers as possible, then decide which one I think is the best fit for me. With an exploding offer, the company is trying to get me to sign the contract by a certain date so that I stop looking for other, potentially much better options.
                      <br />
                      <br />
                      Keep in mind that I wouldn‚Äôt start any of these jobs for a minimum of eight or nine months at this point. If the deadline to accept the offer was, say, a month before I started working, then my case that it was a persuasion tool would be significantly weakened. But that is certainly not the case here.
                      <br />
                      <br />
                      I can hear you thinking, ‚ÄúMark, they probably just wanted a decision by that date so that they could know their recruiting numbers!‚Äù
                      <br />
                      <br />
                      This is not a new argument. First of all, on some level that is a valid concern. Businesses need to gauge their recruitment numbers so that they can hit their quotas, especially during peak recruitment periods. However, this is already a messy problem. Every year, people get offers and turn companies down (without reneging, they just have better opportunities). Almost every company therefore has some kind of policy to extend more offers than they expect to be accepted, or a contingency plan like ‚Äúif person A declines, let‚Äôs offer person B.‚Äù But that‚Äôs okay! Their employees are being paid to manage these problems, and it‚Äôs always a messy situation no matter what. At worst, not setting these arbitrary deadlines makes the situation a little messier for the company to deal with.
                      <br />
                      <br />
                      There is a difference between if I had gone to J.P. Morgan Chase & Co. and said ‚Äúhi, I will have an answer for you by October 1st‚Äù, and them giving me the offer with this ‚Äúexploding‚Äù clause attached to it. In the latter case, they are asserting that they are in control and will dictate all of the terms in the agreement.
                      <br />
                      <br />
                      Recall that different companies have vastly different recruitment timelines and that the 
                      <b>
                        first
                      </b>
                       offer someone receives isn‚Äôt guaranteed to be the 
                      <b>
                        best
                      </b>
                      . With J.P. Morgan Chase & Co., I asked for an extension of the October 1st deadline, and was given a two-week extension until October 15th from the recruiter. Again, I didn‚Äôt say, ‚Äúcan I have an extension until October 15th?‚Äù Rather, the recruiter gave me that arbitrary deadline.
                      <br />
                      <br />
                      In addition, by making excuses for why it is okay for employers to arbitrarily rush such an important decision, we are both absolving them of blame and closing the door to the possibility that this could actually be an underhanded tactic on their part.
                      <br />
                      <br />
                      By October 15th, I was still interviewing and applying for other positions, but I wanted to make sure that I had a job when I graduated no matter what. I had not considered whether or not J.P. Morgan Chase & Co.‚Äôs offer was the best offer, because I knew I wasn‚Äôt done job searching at that point. Because they gave me the arbitrary deadline of October 15th, and I wanted to guarantee that I had a job when I graduated, I accepted it.
                      <br />
                      <br />
                      Around the same time, I was in talks with Learning A-Z about a return offer. I found them honorable in the fact that they didn‚Äôt give me an exploding offer - they just told me that they would prefer to know by a certain date for recruitment purposes. I was happy to respect that request.
                      <br />
                      <br />
                      RetailMeNot gave me an offer in late October, and gave me an exploding deadline in November. It wasn‚Äôt until November 27th that Amazon offered me - but again, they gave me an exploding deadline of January 3rd.
                      <br />
                      <br />
                      You can see how circumstances can make it tough to balance doing what‚Äôs best for the company (like never reneging) and what‚Äôs best for yourself. The important choice is this: if I had received all four offers at the exact same time, what company would I have chosen? My answer was Amazon, so I was willing to renege in order to make the right decision for myself (and, most likely, the company. Who wants an employee that would rather be somewhere else?). It wasn‚Äôt ideal, but reneging was a tool that I used to make the right decision. 
                      <b>
                        Had I simply accepted the first offer that I received and been unwilling to renege, I would have made the wrong decision.
                      </b>
                      <br />
                      <br />
                      <h5
                        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
                      >
                        Companies Looking Out For Their Own Best Interests, Part Two
                      </h5>
                      Employers benefit from being the ones to set the terms in these employment negotiations. If I was a little more short-sighted or if I decided that I was morally against reneging, I would have either 1) accepted the offer and stopped looking for others or 2) turned it down and taken the chance that I wasn‚Äôt employed upon graduation. Neither of these were in my best interest. Someone thinking in the short-term might think that this tactic is in the best interest of the company. It is not.
                      <br />
                      <br />
                      Exploding offers are not in the best interest of either side of the negotiation in the long run. As I alluded to earlier, who wants employees at their company who don‚Äôt want to be there?
                      <br />
                      <br />
                      Imagine a hypothetical where every employee that you hire has their full list of potential offers to accept, but they choose your company. No rushing them, no mismatched deadlines or underhanded tactics. Now you know that, at the very least, your company is full of people who wanted to be there. In and of itself, that can be a strong basis for a powerful, exciting company culture. That‚Äôs the ideal situation. So how can we move closer to that ideal?
                      <br />
                      <br />
                      Imagine a hypothetical where a star high school basketball player, say Lebron James (he never went to college), is being recruited by colleges. A smaller school, say Cornell University, gives him a scholarship offer during his sophomore year of high school. The catch is, they only give him three weeks to decide. Now, as a sophomore in high school, he knows he is pretty good. But he‚Äôs not quite sure if other schools will want him, and he doesn‚Äôt believe that he can risk not going to college on scholarship, given his current financial situation. So he accepts the offer.
                      <br />
                      <br />
                      Now you decide. Was Cornell really looking out for his best interests in that hypothetical situation? Don‚Äôt you think it might have been better if he was able to collect all of his offers, weigh his options, then pick the best one?
                      <br />
                      <br />
                      Imagine how grumpy he would have been when he arrived at Cornell, knowing he made the wrong decision but didn‚Äôt want to back out. Would that really have been the best in the long term for Cornell either?
                      <br />
                      <br />
                      If you have to use extracurricular tactics to force your potential employees to join your company, you are letting everyone down. You are letting the potential employees down because you aren‚Äôt looking out for their best interests. You are letting yourself down because you are admitting that, given all the potential places that Potential Employee #1 could get offers from, they would not choose you. So, as the company, you do things like: setting an arbitrary deadline for them to accept your offer, threaten to rescind it, making them feel bad for reneging after you accepted based on their arbitrary deadlines. You could even argue that setting the salary and terms of employment fall into the same category (why can‚Äôt I say, ‚ÄúI will accept if you give me $80,000 a year‚Äù and then 
                      <b>
                        they
                      </b>
                       can counter if they want, not vice versa?) but that‚Äôs a different discussion altogether.
                      <br />
                      <br />
                      A popular way that employers fight back is by threatening to or actually blacklisting candidates who renege across their industry. This is common in more traditional fields like banking and law, where if Potential Employee #1 decides to renege on an offer from Employer #1, Employer #1 will tell Employer #2 and every other popular employer not to hire Potential Employee #1. Again, this is a negotiation tactic. It‚Äôs a threat that, ‚Äúif you step out of line, we will ruin your career‚Äù and ‚Äúif you step out of line, we will make an example out of you so that nobody else does‚Äù. They are looking out for their own best interests.
                      <br />
                      <br />
                      These are negotiation tactics that the employers think are in their own best interest. Do not confuse their best interests with your own.
                      <br />
                      <br />
                      <h5
                        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
                      >
                        Shifting The Blame To Employees
                      </h5>
                      One of the biggest issues I have is the tendency for us to shift the blame back to the employees. I believe that chastising your friend for reneging isn‚Äôt just incorrect; it‚Äôs morally wrong. It allows the employer to escape the blame while essentially self-policing something that is in the best interest of the company.
                      <br />
                      <br />
                      This is a similar tactic that the government and many malicious corporations have been using forever, where they get ‚Äúthe people‚Äù to fight amongst themselves while standing to the side, twiddling their thumbs and lining their pockets. Resist the temptation to admonish other employees for using any tool that is in their best interest.
                      <br />
                      <br />
                      <h5
                        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
                      >
                        Call To Action
                      </h5>
                      I am asking you to consider ways in which people who seem to be looking out for your best interests (‚Äúbut the recruiter was so nice to me!‚Äù or ‚Äúthey really seemed to care about me!‚Äù) might actually not be. It can take a critical eye and a few deep breaths to see past the smoke and mirrors. That‚Äôs the way they want it - it‚Äôs in their best interest. It‚Äôs hard to fight something you don‚Äôt even know exists.
                      <br />
                      <br />
                      Next time some entity, employer or not, decides that they are going to lay down terms for you before you ever get a chance to say anything, stop for a second. Ask yourself, what are they doing here that is only in their best interest? What is in mine?
                      <br />
                      <br />
                      I make the assertion that arbitrary exploding deadlines, setting the terms of contracts, guilting potential employees into accepting, having ‚Äúnice recruiters‚Äù, threatening to or actually blacklisting potential employees who renege, and many more are leverage tactics by employers. Obviously this is only the tip of the iceberg, but I present these examples to illustrate that we need to be on the lookout for what employers are doing that might not be in our best interest, no matter how they seem at first glance.
                      <br />
                      <br />
                      I also make the case that, as long as employers are going to do what is in their best interest, you must also take steps to do the same for yourself. An employer wouldn‚Äôt hesitate to fire you, rescind your offer, reduce your pay and beyond if that fits their best interest. You must be prepared to use the tools (like reneging) at your disposal to fight for your own best interests. No one else will.
                    </p>
                  </p>
                  <p
                    className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-paragraph"
                  >
                    <i>
                      August 31st, 2020
                    </i>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="InteractionCard-root-4"
  >
    <div
      className="MuiPaper-root MuiCard-root MuiPaper-elevation8 MuiPaper-rounded"
    >
      <a
        className="gone"
        href="#interaction_card"
        name="interaction_card"
      >
        Interaction Card
      </a>
      <div
        className="MuiCardContent-root"
      >
        <div />
        <div
          className="LikeButton-root-5"
        >
          <div
            className="LikeButton-likeButtonWrapper-6"
          >
            <button
              aria-describedby={null}
              className="MuiButtonBase-root MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              title="Click to like page"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <div
              className="MuiPaper-root MuiCard-root LikeButton-captionRoot-7 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <span
                className="MuiTypography-root LikeButton-captionText-8 MuiTypography-caption MuiTypography-colorInherit"
                size="small"
              >
                0
                 likes
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
